src/nwb2bids/_converters/_dataset_converter.py

```diff
@@ -112,24 +147,62 @@ def from_nwb_paths(
         -------
         An instance of DatasetConverter.
         """
-        session_converters = SessionConverter.from_nwb_paths(nwb_paths=nwb_paths)
+        try:
+            session_converters = SessionConverter.from_nwb_paths(nwb_paths=nwb_paths)
+
+            dataset_description = None
+            if additional_metadata_file_path is not None:
+                dataset_description = DatasetDescription.from_file_path(file_path=additional_metadata_file_path)
+
+            session_messages = [
+                message
+                for session_converter in session_converters
+                for message in session_converter.messages
+                if session_converter.messages is not None
+            ]
 
-        dataset_description = None
-        if additional_metadata_file_path is not None:
-            dataset_description = DatasetDescription.from_file_path(file_path=additional_metadata_file_path)
+            dataset_converter = cls(session_converters=session_converters, dataset_description=dataset_description)
+            dataset_converter._internal_messages = session_messages
+            return dataset_converter
+        except:  # noqa
+            _internal_messages = [
+                InspectionResult(
+                    title="Failed to initialize converter on local NWB files",
+                    reason=(
+                        "An error occurred while executing `DatasetConverter.from_nwb_paths`."
+                        f"\n\n{traceback.format_exc()}"
+                    ),
+                    solution="Please raise an issue on `nwb2bids`: https://github.com/con/nwb2bids/issues.",
+                    category=Category.INTERNAL_ERROR,
+                    severity=Severity.ERROR,
+                )
+            ]
 
-        dataset_converter = cls(session_converters=session_converters, dataset_description=dataset_description)
+        dataset_converter._internal_messages = []
```

Comment:
Shouldn't this be adding the messages instead of setting the messages to an empty list?

Reply:
Yes